multimodal/src/autogluon/multimodal/constants.py

```diff
@@ -11,6 +11,8 @@
 MULTICLASS = "multiclass"
 REGRESSION = "regression"
 ZERO_SHOT = "zero_shot"
+ZERO_SHOT_SENTENCE_SIMILARITY = "zero_shot_sentence_similarity"
+ZERO_SHOT_ALL = [ZERO_SHOT, ZERO_SHOT_SENTENCE_SIMILARITY]
```

Comment:
If we go fine-grained, take image classification as an example, we will have `image_classification` and `zero_shot_image_classification`.

For `zero_shot_image_classification`, it will be mainly used for CLIP, LIT kind of models to do inference without training. 

For `image_classification`, it will have three use cases: direct inference, finetuning, and training.

Each use case is different.

(1) zero_shot_image_classification: require both image and text input, no model change, no training, no training data needed, pretrained=True

(2) direct inference: require image as input, no model change, no training, no training data needed, pretrained=True

(3) finetuning: require image as input, has model change (new head), has training and need training data, pretrained_backbone=True

(4) training: require image as input, has model change (new head), has training and need training data, pretrained_backbone=True or you can train from scratch

Other CV tasks are similar. I'm not sure about other applications, but we can discuss problem_types, zero_shot flag, pretrained flag based on these use cases first.

Reply:
For text part, we also have a similar scenario like in (1) when the model structure is two tower. Currently user have to extract the 2 embeddings of paired input and then do cosine similarity themselves (like the example in this PR).  I think we may distinguish (1) and (2) by using predict() in (1) and extract_feature() in (2), while the flag zero_shot distinguishes (1)(2) and (3)(4).